crates/datasources/src/clickhouse/mod.rs

```diff
@@ -117,32 +150,18 @@ impl ClickhouseAccessState {
                     SqlType::Float64 => DataType::Float64,
                     SqlType::String | SqlType::FixedString(_) => DataType::Utf8,
                     // Clickhouse has both a 'Date' type (2 bytes) and a
-                    // 'Date32' type (4 bytes). I think they're both represented
-                    // with this one variant.
+                    // 'Date32' type (4 bytes). The library doesn't support
+                    // 'Date32' type yet.
+                    // TODO: Maybe upstream support for Date32?
                     SqlType::Date => DataType::Date32,
                     SqlType::DateTime(DateTimeType::DateTime32) => {
-                        DataType::Timestamp(TimeUnit::Second, None)
+                        DataType::Timestamp(TimeUnit::Nanosecond, None)
```

Comment:
32 bits of nanoseconds is not very much time. this might be right, but it _looks_ wrong, and the comment doesn't clarify enough.

are there tests over this?

Reply:
So, we essentially convert everything to `chrono::NaiveDateTime` and get `seconds` OR `nanos` from there and we store it in 64 bit (even though clickhouse returns data as 32 bits here).

Reply:
I would think that this would break if we/when we support the write path.

Reply:
What “write path”?

Reply:
like, inserting into clickhouse: if we change the types, round-tripping data will be hard (and also if their underlying type is in seconds, things could overflow in glaredb that wouldn't overflow in the source.)